impl/maven-cli/src/main/java/org/apache/maven/cling/invoker/mvnsh/builtin/BuiltinShellCommandRegistryFactory.java

```diff
@@ -130,53 +129,140 @@ public String name() {
             return "Builtin Maven Shell commands";
         }
 
-        private List<Completers.OptDesc> commandOptions(String command) {
+        private void executeCmnd(List<String> args) throws Exception {
+            ProcessBuilder builder = new ProcessBuilder();
+            List<String> processArgs = new ArrayList<>();
+            if (Os.IS_WINDOWS) {
+                processArgs.add("cmd.exe");
+                processArgs.add("/c");
+            } else {
+                processArgs.add("sh");
+                processArgs.add("-c");
+            }
+            processArgs.add(String.join(" ", args));
```

Comment:
we should add each item as new argument  ... not a one arument with spaces

Reply:
This is shell `-c` and it must be one argument (space separated)